client/dialer.go

```diff
@@ -56,12 +56,12 @@ func (c *HostClient) makeDialer(req Request) transport.Dialer {
 	//set https tls config
 	switch reqType {
 	case requestDirectHTTP:
-		return dialerWrapper(transport.Dial(req.HostInfo().HostWithPort()))
+		return dialerWrapper(transport.Dial(req.TargetWithPort()))
 	case requestDirectHTTPS:
 		if c.tlsServerConfig == nil {
-			c.tlsServerConfig = cert.MakeClientTLSConfig(req.HostInfo().HostWithPort(), req.TLSServerName())
+			c.tlsServerConfig = cert.MakeClientTLSConfig(req.TargetWithPort(), req.TLSServerName())
```

Comment:
req.TargetWithPort() is a misleading for cert.MakeClientTLSConfig, try `cert.MakeClientTLSConfig("", req.TLSServerName())` instead

modify the `cert.MakeClientTLSConfig` if necessary